indexer/src/main/java/au/org/aodn/esindexer/service/GeoNetworkServiceImpl.java

```diff
@@ -107,18 +118,81 @@ public String findGroupById(String uuid) throws IOException {
+        if(optRelated.isPresent()) {
+            Map<String, ?> node = optRelated.get();
+            if(node.containsKey(THUMB_NAILS) && node.get(THUMB_NAILS) instanceof List<?> thumbnails) {
+                // Always use the first item, not sure if it is good?
+                if(!thumbnails.isEmpty() && thumbnails.get(0) instanceof Map<?, ?> thumbnail) {
+                    if(thumbnail.containsKey(URL) && thumbnail.get(URL) instanceof Map<?,?> link) {
+                        return link.entrySet()
+                                .stream()
+                                .findFirst()
+                                .map(i -> {
+                                    LinkModel linkModel = LinkModel.builder().build();
+                                    linkModel.setType("image");
+                                    linkModel.setRel("thumbnail");
+                                    linkModel.setHref(i.getValue().toString());
+                                    return linkModel;
+                                });
+                    }
+                }
+            }
+        }
+        return Optional.empty();
+    }
+    /**
+     * Return a link to the logo if exist.
+     *
+     * @param uuid - UUID of record
+     * @return - Link if logo exist
+     */
+    @Override
+    public Optional<LinkModel> getLogo(String uuid) {
+        Optional<Map<String, Object>> optAdditionalInfo = getRecordExtraInfo(uuid);
+        if(optAdditionalInfo.isPresent()) {
+            // We iterate logos link and add it to STAC
+            Map<String, Object> additionalInfo = optAdditionalInfo.get();
+            if(additionalInfo.containsKey(SUGGEST_LOGOS)) {
+                if(additionalInfo.get(SUGGEST_LOGOS) instanceof List) {
+                    return ((List<?>) additionalInfo.get(SUGGEST_LOGOS))
+                            .stream()
+                            .map(p -> (p instanceof String) ? (String) p : null)
+                            .filter(Objects::nonNull)
+                            .filter(i -> urlUtils.checkUrlExists(i))
+                            .findFirst()        // We only pick the first reachable one
+                            .map(i -> {
+                                LinkModel linkModel = LinkModel.builder().build();
+                                linkModel.setHref(i);
+                                // Geonetwork always return png logo
+                                linkModel.setType("image/png");
+                                linkModel.setRel("icon");
+                                linkModel.setTitle("Suggest icon for dataset");
+                                return linkModel;
+                            });
+                }
+            }
+        }
+        return Optional.empty();
+    }
+
+    protected Optional<Map<String, Object>> getRecordExtraInfo(String uuid) {
         Map<String, Object> params = new HashMap<>();
         params.put(UUID, uuid);
 
         try {
```

Comment:
there is `try-catch` block here but should we use `throw new ... exception message` to keep the logs clean instead of trying to `return` inside `catch`? Eventually at the end, line `207`,  there is another `return Optional.empty()`

Reply:
I do not want this method to throw exception, if error return Optional.empty()